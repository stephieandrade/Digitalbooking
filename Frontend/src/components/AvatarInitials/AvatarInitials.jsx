import { useEffect, useState } from "react";

import "./avatar-initials.scoped.css";

const AvatarInitials = ({ name }) => {
  const [initials, setInitials] = useState("");
  useEffect(() => {
    setInitials(getInitials(name));
  }, [name]);

  function getInitials(name) {
    if (name === undefined) return "";
    const [firstName, lastName] = name.split(" ");
    if (firstName && lastName) {
      return `${firstName.charAt(0)}${lastName.charAt(0)}`.toUpperCase();
    }
    return firstName.charAt(0).toUpperCase();
  }

  return (
    <div className="initials">
      <p>{initials}</p>
    </div>
  );
};

export default AvatarInitials;
