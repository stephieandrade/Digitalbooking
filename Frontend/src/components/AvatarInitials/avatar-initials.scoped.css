/* Iniciales en mobile */
div.initials {
  width: 37px;
  height: 37px;
  min-width: 37px;
  background-color: var(--five-color);
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
}

/* Texto en mobile*/
p {
  color: var(--two-color);
  font-size: 20px;
  font-weight: 700;
}

/* Texto y color del fondo de iniciales en Tablet/Desktop*/
@media screen and (min-width: 768px) {
  div.initials {
    background-color: var(--two-color);
  }
  p {
    color: var(--five-color);
  }
}
