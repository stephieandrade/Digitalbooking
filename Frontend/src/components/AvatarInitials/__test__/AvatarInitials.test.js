import React from "react";
import AvatarInitials from "../AvatarInitials";
import { mount, shallow } from "enzyme";

//const runAllPromises = () => new Promise(setImmediate);

describe("Pruebas en <AvatarInitials/>", () => {
  test("Dos iniciales en mayúscula para dos nombres", async () => {
    const nombre = "diego fernando";
    const wrapper = mount(<AvatarInitials name={nombre} />);
    //await runAllPromises();
    expect(wrapper).toMatchSnapshot();
  });

  test("Dos iniciales en mayúscula para tres nombres", async () => {
    const nombre = "diego fernando díaz";
    const wrapper = mount(<AvatarInitials name={nombre} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Una inicial en mayúscula para un nombre", async () => {
    const nombre = "diego";
    const wrapper = mount(<AvatarInitials name={nombre} />);
    expect(wrapper).toMatchSnapshot();
  });

  it ('tiene una etiqueta p', () => {
    const component = shallow(<AvatarInitials/>);
    const node = component.find('p');
    expect(node.length).toEqual(1);

   });


});
