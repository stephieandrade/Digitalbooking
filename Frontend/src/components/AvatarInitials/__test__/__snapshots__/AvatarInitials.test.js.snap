// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AvatarInitials/> Dos iniciales en mayúscula para dos nombres 1`] = `
<div
  className="initials"
>
  <p
    className="text"
  >
    DF
  </p>
</div>
`;

exports[`Pruebas en <AvatarInitials/> Dos iniciales en mayúscula para tres nombres 1`] = `
<div
  className="initials"
>
  <p
    className="text"
  >
    DF
  </p>
</div>
`;

exports[`Pruebas en <AvatarInitials/> Una inicial en mayúscula para un nombre 1`] = `
<div
  className="initials"
>
  <p
    className="text"
  >
    D
  </p>
</div>
`;
